torch/csrc/cuda/nccl.cpp

```diff
@@ -123,6 +136,57 @@ static inline void NCCL_CHECK(ncclResult_t result) {
   NCCL_CHECK(from_nccl_result(result));
 }
 
+static inline void NCCL_CHECK_NONBLOCKING(ncclResult status, ncclComm_t comm) {
+#ifdef NCCL_HAS_COMM_NONBLOCKING
+  ncclResult_t result = to_nccl_result(status);
+  while (result == ncclInProgress) {
+    ncclCommGetAsyncError(to_nccl_comm(comm), &result);
```

Comment:
I wonder if `to_nccl_comm(comm)` is needed here.

Here is definition of `to_nccl_comm`:
```
ncclComm_t to_nccl_comm(torch::cuda::nccl::ncclComm_t var) {
  return reinterpret_cast<ncclComm_t>(var);
}
```

It seems to me `comm` is already a `ncclComm_t` (the one defined by NCCL).

Side note: 
We should remove the duplicated `ncclComm_t` definition in `torch::cuda::nccl`. It is making things complicated.
It is out of scope of this PR. We can do that later.